src/main/java/com/gluonhq/Main.java

```diff
@@ -167,6 +169,15 @@ private Button actionButton(Ikon ikon, Action action) {
         return button;
     }
 
+    private ToggleButton createToggleButton(Ikon ikon, Function<ObjectProperty<Boolean>, DecorateAction<Boolean>> function) {
+        final ToggleButton fontWeight = new ToggleButton();
+        FontIcon icon = new FontIcon(ikon);
+        icon.setIconSize(20);
+        fontWeight.setGraphic(icon);
+        function.apply(fontWeight.selectedProperty().asObject());
```

Comment:
This should be fixed now. Can you re-test?

Reply:
Applying background color seems to fail now. Once applied it doesn't remain

Reply:
I see the issue. We currently have TextDecoration on a view model, which mostly represents the decoration currently selected by the user or applied on the previous piece depending on the position of the cursor.

If we select text spanning multiple pieces and apply a new TextDecoration then we cannot do so without taking into consideration the TextDecoration already applied on these individual pieces.

Reply:
Right, if you select Open Text, the text has two lines, blue and red. Select all content, and press Bold. All turns bold _and_ red, which is unexpected: The font weight should be applied to all pieces, keeping the rest of their existing decoration attributes (i.e. blue and red foreground colors). 

As is now, if we make a selection of multiple pieces with multiple decorations, the toolbar is only able to keep a single decoration, and when pressing any of the control buttons, that full decoration is the one applied to all pieces. 
This works fine when the selection only spans one piece or pieces with the same decoration, as all the buttons update with such decoration, and only one attribute is changed with the button that is pressed. 

We might need to modify the decoration builder in order to change _only_ one attribute at a time: the decorate action should retrieve the existing decoration for each piece, instead of reading the full decoration from the toolbar, and then apply the change on that single attribute for all these pieces.

Does this make sense?

Reply:
So this:
```
TextDecoration newTextDecoration = TextDecoration.builder().fromDecoration(viewModel.getTextDecoration()).foreground(nv).build();
                 ACTION_CMD_FACTORY.decorateText(newTextDecoration).apply(viewModel);
```
should probably something like:
```
TextDecoration newTextDecoration = TextDecoration.builder().emptyDecoration().foreground(nv).build();
                 ACTION_CMD_FACTORY.decorateText(newTextDecoration).apply(viewModel);
                 
```                 
where `emptyDecoration` is just a decoration with all null values, so only foreground is changed...